pkg/cmd/roachtest/test_runner.go

```diff
@@ -913,6 +917,47 @@ fi'`
 	}
 }
 
+// getGoCoverArtifacts retrieves the go coverage artifacts for the test.
+// If there's an error, oh well, don't do anything rash like fail a test
+// which already passed.
+func getGoCoverArtifacts(ctx context.Context, l *logger.Logger, c *clusterImpl, t test.Test) {
+	g := ctxgroup.WithContext(ctx)
+	fetchNode := func(node int) func(context.Context) error {
+		return func(ctx context.Context) error {
+			testCmd := `'GOCOVER_ARTIFACTS="` + t.GoCoverArtifactsDir() + `"
+if [[ -d "${GOCOVER_ARTIFACTS}" ]]; then
+    echo true
+elif [[ -e "${GOCOVER_ARTIFACTS}" ]]; then
+    ls -la "${GOCOVER_ARTIFACTS}"
+    exit 1
+else
+    echo false
+fi'`
+			result, err := c.RunWithDetailsSingleNode(ctx, t.L(), c.Node(node), "bash", "-c", testCmd)
```

Comment:
Nit: `bash -c` shouldn't be needed. `RunWithDetailsSingleNode` already adds `bash -c`.